t_stress/test_header_leak.py

```diff
@@ -0,0 +1,142 @@
+__author__ = "Tempesta Technologies, Inc."
+__copyright__ = "Copyright (C) 2022 Tempesta Technologies, Inc."
+__license__ = "GPL2"
+
+import random
+import socket
+import ssl
+import string
+from threading import Thread
+
+import h2
+
+from framework import tester
+from helpers import tf_cfg
+
+
+def randomword(length):
+    letters = string.ascii_lowercase
+    return "".join(random.choice(letters) for i in range(length))
+
+
+class TestH2HeaderLeak(tester.TempestaTest):
+    backends = [
+        {
+            "id": "nginx",
+            "type": "nginx",
+            "status_uri": "http://${server_ip}:8000/nginx_status",
+            "config": """
+pid ${pid};
+worker_processes  auto;
+events {
+    worker_connections   1024;
+    use epoll;
+}
+http {
+    keepalive_timeout ${server_keepalive_timeout};
+    keepalive_requests 10;
+    tcp_nopush       on;
+    tcp_nodelay      on;
+    error_log /dev/null emerg;
+    access_log off;
+    server {
+        listen        ${server_ip}:8000;
+        location / {
+            return 200 'foo';
+        }
+        location /nginx_status {
+            stub_status on;
+        }
+    }
+}
+""",
+        }
+    ]
+
+    tempesta = {
+        "config": """
+        cache 0;
+
+        keepalive_timeout 1000;
+
+        listen 443 proto=h2;
+
+        tls_match_any_server_name;
+
+        srv_group default {
+            server ${server_ip}:8000;
+        }
+
+        vhost tempesta-tech.com {
+           tls_certificate ${tempesta_workdir}/tempesta.crt;
+           tls_certificate_key ${tempesta_workdir}/tempesta.key;
+           proxy_pass default;
+        }
+        """
+    }
+
+    def test(self):
```

Comment:
I think you should check the memory consumption in this test. For example:

1. save current the memory consumption to variable;
2. send requests;
3. compare current the memory consumption with variable from p.1;

Currently this test check nothing and it is not possible to find the problem (OOM) for different virtual machines.

And I still think you can use DeproxyClientH2 here (with Nginx as backend) because a response is very small and I don't think it won't affect testing.

This test is very similar to the [hpack bomb test](https://github.com/tempesta-tech/tempesta-test/blob/ab21520ff7e638c657820a5dd60af751f7c1bbd2/http2_general/test_h2_hpack.py#L931)